python/ray/widgets/util.py

```diff
@@ -102,6 +102,43 @@ def wrapped(*args, **kwargs):
     return wrapper
 
 
+@DeveloperAPI
+def ensure_ipywidgets_dep(version: str) -> Callable[[F], F]:
+    """Generate a decorator which checks for a soft ipywidgets dependency.
+
+    This is a convencience function separate from `ensure_notebook_deps` because
+    of its custom missing and outdated messages, which suggest the user restart the
+    notebook server after installation/upgrade.
+
+    Args:
+        version: Version of ipywidgets required.
+
+    Returns:
+        Wrapped function. Guaranteed to be safe against the specified ipywidgets
+        version.
+    """
+    text = (
+        "Run `pip install {}ipywidgets`, then restart "
+        "the notebook server for rich notebook output."
+    )
+
+    if in_notebook():
+        return ensure_notebook_deps(
+            ["ipywidgets", version],
+            missing_message=text.format(""),
```

Comment:
Could we still print the string repr if the notebook deps aren't found? It seems like the current code will fail to print any repr if the deps are unavailable.

Reply:
updated